activesupport/lib/active_support/notifications/instrumenter.rb

```diff
@@ -143,28 +143,30 @@ def parent_of?(event)
 
       private
         def now
-          Concurrent.monotonic_time
+          Process.clock_gettime(Process::CLOCK_MONOTONIC, :float_millisecond)
         end
 
         begin
-          Process.clock_gettime(Process::CLOCK_THREAD_CPUTIME_ID)
+          Process.clock_gettime(Process::CLOCK_THREAD_CPUTIME_ID, :float_millisecond)
 
           def now_cpu
-            Process.clock_gettime(Process::CLOCK_THREAD_CPUTIME_ID)
+            Process.clock_gettime(Process::CLOCK_THREAD_CPUTIME_ID, :float_millisecond)
           end
         rescue
           def now_cpu
-            0
+            0.0
           end
         end
 
-        if defined?(JRUBY_VERSION)
+        begin
+          GC.stat(:total_allocated_objects)
+        rescue ArgumentError # Likely on JRuby
```

Comment:
Thanks for this change, TruffleRuby also cannot report the number of allocated objects (not exposed on JVM) so this might be helpful :)

Reply:
Welcome.